### PR TITLE
fix: Update build

### PR DESCRIPTION
**Pull request by rygine**
# Summary

This update adds tree-shaking to the JS build so that unused code paths are removed. This removes references to unused files. The terser plugin was also added to decrease the size of the source.

- Upgraded node version in `.node-version`
- Upgraded yarn
- Removed `package-lock.json`
- Fixed VSCode `rust-analyzer` settings
- Updated GitHub workflows
- Updated README
- Updated rollup config
- Added `wasm-pack` and `@rollup/plugin-terser` dependencies
- Renamed `package.js` to `sqlite3-diesel.js`
- Added test scripts
- Added tree-shaking and terser plugin for output

## Release Notes

- **New Features**
	- Updated workflow configurations for better performance and clarity in CI/CD processes.
	- Introduced new test scripts for comprehensive browser testing, including headless modes.

- **Bug Fixes**
	- Corrected paths and dependencies in the README for improved clarity and consistency.

- **Documentation**
	- Enhanced README with clearer instructions and corrected examples.

- **Chores**
	- Updated Node.js version and Yarn configuration for improved compatibility and performance.
	- Streamlined JavaScript code for better readability and maintainability.

- **Style**
	- Modernized JavaScript syntax in the async proxy file for enhanced clarity.

### CODE REVIEW

File: .github/workflows/lint-pr.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ permissions:
 jobs:
   main:
     name: Validate PR title
-    runs-on: ubuntu-latest
+    runs-on: warp-ubuntu-latest-x64-16x
```
Comment (insipx) [inline .github/workflows/lint-pr.yml:14]: unfortunately I dont think the warp builds work in this repo, i don't know if theres something that needs to change in the warp service but I don't have access there

**Review by insipx — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-pr.yml (hunk 1 above)
Comment (rygine) [reply .github/workflows/lint-pr.yml:14]: this workflow looks like it ran fine, but the test one didn't. hmmm. does it need to run on a mac runner? doesn't look like it's running the tests in safari. maybe switching the mac runner to ubuntu will work.

Comment (rygine) [reply .github/workflows/lint-pr.yml]: ok, switched the runners to `ubuntu-latest`. no idea why just the 1 warp runner didn't work.

**Comment by rygine** [comment]
@insipx is there anything that we need to do in this crate for `sqlite3.wasm` to be part of the files copied over or is that something we need to do in `xtask`?

with these changes, `sqlite3.wasm` is still referenced in the code and is required for production builds, but it's never actually accessed.

**Comment by insipx** [comment]
> @insipx is there anything that we need to do in this crate for sqlite3.wasm to be part of the files copied over or is that something we need to do in xtask?

with these changes, sqlite3.wasm is still referenced in the code and is required for production builds, but it's never actually accessed.


@rygine need to copy `sqlite3.wasm` in the xtask in `libxmtp` for now. I'll create a ticket for a better solution sometime, maybe with a patch to the sqlite repo

**Review by insipx — APPROVED** [review]

**Pull request merged**